numpy/lib/function_base.py

```diff
@@ -4294,8 +4303,21 @@ def percentile(a, q, axis=None, out=None,
         return r
 
 
-def _percentile(a, q, axis=None, out=None,
-                overwrite_input=False, interpolation='linear', keepdims=False):
+def _quantile_is_valid(q):
+    # avoid expensive reductions, relevant for arrays with < O(1000) elements
+    if q.ndim == 1 and q.size < 10:
+        for i in range(q.size):
+            if q[i] < 0.0 or q[i] > 1.0:
+                return False
+    else:
+        # faster than any()
+        if np.count_nonzero(q < 0.0) or np.count_nonzero(q > 1.0):
```

Comment:
Could probably use `|` here, something like `np.count_nonzero((q < 0.0) | (q > 1.0))`, or even in place. Just a note for the separate PR.

Reply:
Doing the `or` vectorized is algorithmically slower - `n` or operations vs `1`.